crates/re_viewer_context/src/space_view/dyn_space_view_class.rs

```diff
@@ -44,12 +44,14 @@ pub enum SpaceViewClassLayoutPriority {
 pub trait DynSpaceViewClass {
     /// Name of this space view class.
     ///
-    /// Used for both ui display and identification.
-    /// Must be unique within a viewer session.
-    ///
-    /// TODO(#2336): Display name and identifier should be separate.
+    /// Used for identification. Must be unique within a viewer session.
     fn name(&self) -> SpaceViewClassName;
 
+    /// User-facing name of this space view class.
+    ///
+    /// Used for UI display.
+    fn ui_name(&self) -> &'static str;
```

Comment:
I think the more common name for this in the codebase is `display_name`.
... come to think it's more like the default display name? Because the user can edit the actual display name. From that pov ui_name is probably good as is since it avoid confusing it with the configurable `display_name` of a space view instance 🤔 . Your call :)